src/main/java/com/google/api/codegen/config/FieldConfig.java

```diff
@@ -212,10 +212,14 @@ public FieldConfig withResourceNameConfig(ResourceNameConfig resourceNameConfig)
   public boolean requiresParamTransformation() {
     return getResourceNameConfig() != null
         && getMessageResourceNameConfig() != null
-        && getMessageResourceNameConfig().getResourceNameType() != ResourceNameType.ANY
         && !getResourceNameConfig().equals(getMessageResourceNameConfig());
   }
 
+  public boolean requiresAnyParamTransformation() {
```

Comment:
Do you think it would be better to implement `.equals()` as reference equality, and rely on another method to check whether the string formats are equal? (Or just force the use of `toString()` to check that?) This could be outside the scope of the PR.